mmcls/datasets/stanford_car.py

```diff
@@ -0,0 +1,205 @@
+        'Jeep Compass SUV 2012', 'Lamborghini Reventon Coupe 2008',
+        'Lamborghini Aventador Coupe 2012',
+        'Lamborghini Gallardo LP 570-4 Superleggera 2012',
+        'Lamborghini Diablo Coupe 2001', 'Land Rover Range Rover SUV 2012',
+        'Land Rover LR2 SUV 2012', 'Lincoln Town Car Sedan 2011',
+        'MINI Cooper Roadster Convertible 2012',
+        'Maybach Landaulet Convertible 2012', 'Mazda Tribute SUV 2011',
+        'McLaren MP4-12C Coupe 2012',
+        'Mercedes-Benz 300-Class Convertible 1993',
+        'Mercedes-Benz C-Class Sedan 2012',
+        'Mercedes-Benz SL-Class Coupe 2009',
+        'Mercedes-Benz E-Class Sedan 2012', 'Mercedes-Benz S-Class Sedan 2012',
+        'Mercedes-Benz Sprinter Van 2012', 'Mitsubishi Lancer Sedan 2012',
+        'Nissan Leaf Hatchback 2012', 'Nissan NV Passenger Van 2012',
+        'Nissan Juke Hatchback 2012', 'Nissan 240SX Coupe 1998',
+        'Plymouth Neon Coupe 1999', 'Porsche Panamera Sedan 2012',
+        'Ram C/V Cargo Van Minivan 2012',
+        'Rolls-Royce Phantom Drophead Coupe Convertible 2012',
+        'Rolls-Royce Ghost Sedan 2012', 'Rolls-Royce Phantom Sedan 2012',
+        'Scion xD Hatchback 2012', 'Spyker C8 Convertible 2009',
+        'Spyker C8 Coupe 2009', 'Suzuki Aerio Sedan 2007',
+        'Suzuki Kizashi Sedan 2012', 'Suzuki SX4 Hatchback 2012',
+        'Suzuki SX4 Sedan 2012', 'Tesla Model S Sedan 2012',
+        'Toyota Sequoia SUV 2012', 'Toyota Camry Sedan 2012',
+        'Toyota Corolla Sedan 2012', 'Toyota 4Runner SUV 2012',
+        'Volkswagen Golf Hatchback 2012', 'Volkswagen Golf Hatchback 1991',
+        'Volkswagen Beetle Hatchback 2012', 'Volvo C30 Hatchback 2012',
+        'Volvo 240 Sedan 1993', 'Volvo XC90 SUV 2007',
+        'smart fortwo Convertible 2012'
+    ]
+
+    def __init__(self,
+                 data_prefix: str,
+                 test_mode: bool,
+                 ann_file: Optional[str] = None,
+                 **kwargs):
+        if test_mode:
+            if ann_file is not None:
+                self.test_ann_file = ann_file
+            else:
+                self.test_ann_file = osp.join(
+                    data_prefix, 'devkit/cars_test_annos_withlabels.mat')
+            data_prefix = osp.join(data_prefix, 'cars_test')
+        else:
+            if ann_file is not None:
+                self.train_ann_file = ann_file
+            else:
+                self.train_ann_file = osp.join(data_prefix,
+                                               'devkit/cars_train_annos.mat')
+            data_prefix = osp.join(data_prefix, 'cars_train')
+        super(StanfordCars, self).__init__(
+            ann_file=ann_file,
+            data_prefix=data_prefix,
+            test_mode=test_mode,
+            **kwargs)
+
+    def load_annotations(self):
+        data_infos = []
+        if self.test_mode:
+            data = sio.loadmat(self.test_ann_file)
+        else:
+            data = sio.loadmat(self.train_ann_file)
```

Comment:
Please import `scipy.io` only when using it. The `scipy` package is a little large, and if we only use it in a dataset class, don't make it a required package.

Reply:
Same in the unit tests.